editor/quick_open.cpp

```diff
@@ -224,13 +232,23 @@ void EditorQuickOpen::_notification(int p_what) {
 		case NOTIFICATION_EXIT_TREE: {
 			disconnect("confirmed", callable_mp(this, &EditorQuickOpen::_confirmed));
 		} break;
+		case NOTIFICATION_VISIBILITY_CHANGED: {
+			if (is_visible()) {
+				exclude_addons->set_pressed(ProjectSettings::get_singleton()->get("editor/exclude_addons_quick_open"));
+			}
+		} break;
 	}
 }
 
 void EditorQuickOpen::_theme_changed() {
 	search_box->set_right_icon(search_options->get_theme_icon("Search", "EditorIcons"));
 }
 
+void EditorQuickOpen::_save_exclude_addons() {
```

Comment:
Why is this a project setting but also a checkbox in the dialog? I am against introducing checkboxes to Quick Open dialogs, but a project setting could be fine. That would allow this filter to be done just once when opening the dialog in `_build_search_cache`.

Reply:
Imagine the user doesn't want to see addon files in his/her current project. But there came a time when s/he needs to search for a file from addons (for whatever reason). S/he opens project settings, find the setting, clicks it, opens the Quick Open dialog, finds the file s/he needs, opens the project settings again, finds the setting and clicks it again.
Unless there's a checkbox for than in Quick Open dialog that simplifies that process a lot. Which is the reason i've added it.

Reply:
I agree, it makes sense to expose this in a checkbox in the Quick Open dialogs. Its value should persist across restarts.

Reply:
I made it a project setting since the user might have some project when s/he doesn't want to see addons' files and some projects (like addons s/he's developing) where s/he does. But if editor settings is a better place for that, sure, i can change.

Reply:
@hilfazer I don't imagine Quick Open is the first thing they will use to find it, I imagine they will just use the filesystem dock. But even this unlikely scenario you describe doesn't sound unreasonable. It's certainly preferable over introducing edge case checkboxes in every place, especially Quick Open dialogs. Unnecessary elements like this are a distraction that go against its purpose.